### PR TITLE
feat: onRenderContextMenu context selection

### PR DESCRIPTION
**Pull request by 1pone**
feat: 
Using the context values `selection`, different actions can be taken depending on the path of selection.

fix: 
package.json @lezer/highlight

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -331,10 +331,10 @@ const editor = new JSONEditor({
       }
       ```
 
-  - `onRenderContextMenu(items: ContextMenuItem[], context: { mode: 'tree' | 'text' | 'table', modal: boolean }) : ContextMenuItem[] | undefined`.
+  - `onRenderContextMenu(items: ContextMenuItem[], context: { mode: 'tree' | 'text' | 'table', modal: boolean, selection?: JSONEditorSelection | null }) : ContextMenuItem[] | undefined`.
```
Comment (josdejong) [inline README.md:334]: I think the definition should be without a question mark:

```
, selection: JSONEditorSelection | null }
```

File: package.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "svelte-jsoneditor",
   "description": "A web-based tool to view, edit, format, transform, and validate JSON",
-  "version": "0.21.1",
+  "version": "0.21.2",
```
Comment (josdejong) [inline package.json:4]: Can you leave the version untouched? That will be automatically increased when publishing a new version.

File: pnpm-lock.yaml
Comment (josdejong) [inline pnpm-lock.yaml:1]: Can you please remove the yaml lock file?

File: src/routes/development/+page.svelte [hunk 3]
```diff
@@ -377,6 +378,10 @@
     console.log('onChangeQueryLanguage', newQueryLanguageId)
     queryLanguageId = newQueryLanguageId
   }
+  function onRenderContextMenu(item:ContextMenuItem[], content:RenderMenuContext)  {
+    console.log('onRenderContextMenu',item, content)
```
Comment (josdejong) [inline src/routes/development/+page.svelte:382]: Can you run `npm run format` to fix any linting issues?

**Review by josdejong — CHANGES_REQUESTED** [review]
Thanks, passing the `selection` along with onRenderContextMenu makes sense!

I made a couple of inline comments, can you have a look at those?

**Comment by 1pone** [comment]
done

**Review by josdejong — APPROVED** [review]

**Comment by josdejong** [comment]
Thanks for the updates! Merging now.

**Comment by josdejong** [comment]
Published now in `v0.21.2`. Thanks again.

**Pull request merged**